zk_test.go

```diff
@@ -1195,6 +1184,33 @@ func expectLogMessage(t *testing.T, logger *testLogger, pattern string) {
 	}
 }
 
+func requireMinimumZkVersion(t *testing.T, minimum string) {
+	t.Helper()
+
+	actualVersionStr, ok := os.LookupEnv("ZK_VERSION")
+	if !ok {
+		t.Skip("did not detect ZK_VERSION from env. skipping test")
+	}
+
+	parseFn := func(v string) (parts []int) {
+		for _, s := range strings.Split(v, ".") {
+			i, err := strconv.Atoi(s)
+			if err != nil {
+				t.Fatalf("invalid version segment: %q", s)
+			}
+			parts = append(parts, i)
+		}
+		return parts
+	}
+
+	minimumV, actualV := parseFn(minimum), parseFn(actualVersionStr)
+	for i, p := range minimumV {
+		if actualV[i] < p {
```

Comment:
This assumes that the number of parts in the actual value is greater than or equal to the number of parts in the expected value, and would panic otherwise.

While there's limited impact to panicking in a unit test context, can this be handled more gracefully?

Reply:
great point, I think more to the spirt of the function, we expect required to be shorter, but we don't want to fail if its longer.  

will update